src/Symfony/Component/Security/Core/Authorization/UserAuthorizationChecker.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization;
+
+use Symfony\Component\Security\Core\Authentication\Token\UserAuthorizationCheckerToken;
+use Symfony\Component\Security\Core\Authorization\Voter\AuthenticatedVoter;
+use Symfony\Component\Security\Core\Exception\InvalidArgumentException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * UserAuthorizationChecker is used to check user authorization without a session.
+ *
+ * @author Nate Wiebe <nate@northern.co>
+ */
+class UserAuthorizationChecker
```

Comment:
We have an interface for `AuthorizationChecker`. Should we do the same for `UserAuthorizationChecker`? This would also allow us to finalize this class.

Reply:
Updated.